src/lib.rs

```diff
@@ -1,10 +1,20 @@
-use std::borrow::Cow;
 use std::collections::HashSet;
 
 /// A set of word pieces.
-pub struct WordPieces(pub HashSet<String>);
+pub struct WordPieces {
+    prefixes: HashSet<String>,
+    suffixes: HashSet<String>,
+}
 
 impl WordPieces {
+    /// Construct new word pieces instance.
+    ///
+    /// The arguments are the prefix and suffix set. The prefix set
```

Comment:
The suffix set should have the suffix-marker removed, right?

Reply:
Whoops, typo. Fixed.